File: Hydrogene/Nougat/UIKit/UIAlertView/UIAlertView/RootViewController.cs
namespace CSharpiOSUIAlertViewDemo
{
	using UIKit;

	[IBObject]
	class RootViewController : UIViewController
	{

    [IBOutlet]
    public UITextField messageField;

		public override id init()
		{
			this = base.initWithNibName("RootViewController") bundle(null);
			if (this != null)
			{
				title = "CSharp UIAlertView Demo";
			}
			return this;
		}


    [IBAction]
		public  void sendMessage(id sender)
		{
			UIAlertView message = new UIAlertView withTitle("Message") message(this.messageField.text) @delegate(this) cancelButtonTitle("OK") otherButtonTitles("More Info", "Even More Info", null);
		  message.show();
		}

    
	  public void alertView(UIAlertView alertView) clickedButtonAtIndex(NSInteger buttonIndex)
    {
        string buttonText = alertView.buttonTitleAtIndex(buttonIndex);
        NSLog("Button Text:%@",buttonText );
        if (buttonText == "OK")
          NSLog("OK");
       
        switch (buttonText)
        {
           case "OK": 
			     {
					   NSLog("OK was selected.");
             break;
           }
        }
    }


	}
}

File: Oxygene/Toffee/UIKit/UIAlertView/UIAlertView/RootViewController.pas
namespace OxygeneiOSUIAlertViewDemo;

interface

uses
  UIKit;
  

type
  [IBObject]
  RootViewController = public class(UIViewController, IUIAlertViewDelegate)
  private
    method alertView(alertView : UIAlertView) clickedButtonAtIndex(buttonIndex: NSInteger);
  public
    method init: id; override;

    [IBOutlet] messageField : UITextField;
    [IBAction] method sendMessage(sender: id);
  end;

implementation

method RootViewController.init: id;
begin
  self := inherited initWithNibName('RootViewController') bundle(nil);
  if assigned(self) then begin
    title := 'Oxygene UIAlertView Demo';
  end;
  result := self;
end;

method  RootViewController.sendMessage(sender: id);
begin
  var message := new UIAlertView withTitle("Message") message(messageField.text) &delegate(self) cancelButtonTitle("OK") otherButtonTitles("More Info", "Even More Info", nil);
  message.show();
end;

method RootViewController.alertView(alertView : UIAlertView) clickedButtonAtIndex(buttonIndex: NSInteger);
begin
   case alertView.buttonTitleAtIndex(buttonIndex) of 
    "OK": NSLog("OK was selected.");
    "More Info": NSLog("More Info was selected.");
    "Even More Info": NSLog("Even More Info was selected.");
   end;
end;

end.
